File: Code/Components/DanpheEMR.ServerModel/InventoryModels/ReturnToVendorModel.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace DanpheEMR.ServerModel
{
   public  class ReturnToVendorModel   {

        [Key]
        public int ReturnToVendorId { get; set; }
        public DateTime ReturnDate { get; set; }
        public int VendorId { get; set; }
        public decimal SubTotal { get; set; }
        public decimal VATTotal { get; set; }
        public decimal DiscountAmount { get; set; }
        public decimal TotalAmount { get; set; }
        public int CreditNoteId { get; set; }
        public int? CreditNotePrintNo { get; set; }
        public string Remarks { get; set; }
        public int CreatedBy { get; set; }
        public DateTime CreatedOn { get; set; }
        public int? ModifiedBy { get; set; }
        public DateTime? ModifiedOn { get; set; }

        [NotMapped]
        public List <ReturnToVendorItemsModel> itemsToReturn { get; set;}
        public int StoreId { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/inventory/return-to-vendor/return-to-vendor.model.ts
import { ReturnToVendorItem } from "./return-to-vendor-items.model";

export class ReturnToVendorModel {
  public ReturnToVendorId: number = 0;
  public ReturnDate: string;
  public VendorId: number = null;
  public Remarks: string = null;
  public SubTotal: number = 0;
  public VATTotal: number = 0;
  public DiscountAmount: number = 0;
  public TotalAmount: number = 0;
  public CreditNoteId: number = null;
  public CreditNotePrintNo: number = null;
  public CreatedOn: string = null;
  public CreatedBy: number = null;
  public ModifiedBy: number = null;
  public ModifiedOn: string = null;
  public itemsToReturn: Array<ReturnToVendorItem> = new Array<ReturnToVendorItem>();
  public StoreId: number;
}
